src/Analysis/GWASUIApp/GWASJob.jsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import {
+  Button, List, Tag,
+} from 'antd';
+import {
+  CheckCircleOutlined,
+  SyncOutlined,
+  CloseCircleOutlined,
+  QuestionCircleOutlined,
+  MinusCircleOutlined,
+  StopOutlined,
+} from '@ant-design/icons';
+import { useQuery } from 'react-query';
+import PropTypes from 'prop-types';
+import { gwasWorkflowPath } from '../../localconf';
+import { headers } from '../../configs';
+import { getPresignedUrl } from '../AnalysisJob';
+
+const GWASJob = ({ workflow }) => {
+  async function handleWorkflowOutput(url) {
+    const response = await fetch(url, { headers }).then((res) => res.json()).then((data) => data);
+    if (response) {
+      getPresignedUrl(JSON.parse(response.outputs.parameters[0].value).did, 'download')
+        .then((res) => {
+          window.open(res, '_blank');
+        });
+    }
+  }
+
+  async function handleWorkflowLogs(url) {
+    window.open(url, '_blank');
+  }
+
+  const getActionButtons = (phase, workflowName) => {
+    const actionButtons = [];
+    let actionUrl;
+    let buttonText;
+    let buttonClickHandler;
+
+    if (phase === 'Succeeded') {
+      actionUrl = `${gwasWorkflowPath}status/${workflowName}`;
+      buttonText = 'download outputs';
+      buttonClickHandler = handleWorkflowOutput;
+    } else if (phase === 'Failed') {
+      actionUrl = `${gwasWorkflowPath}logs/${workflowName}`;
+      buttonText = 'view logs';
+      buttonClickHandler = handleWorkflowLogs;
+    }
```

Comment:
can we log something if phase is not "Succeeded" or "Failed", just in case we have other phases in the future

Reply:
I think the workflow phases are defined here: https://github.com/argoproj/argo-workflows/blob/master/pkg/apis/workflow/v1alpha1/workflow_phase.go#L8-L12

Reply:
saving these statuses to `constants.js`